lib/one_time_password/validator.rb

```diff
@@ -1,18 +1,19 @@
 module OneTimePassword
   class Validator < Base
-    def initialize(code, generated_at)
+    def initialize(code, generated_at = nil)
       @code = code
       @generated_at = generated_at
     end
 
     def valid?
-      code.present? && !wrong_length? && !expired? && !incorrect?
+      code.present? && !wrong_length? && (generated_at && !expired?) && !incorrect?
```

Comment:
- if `generated_at` is `nil` won't `valid?` always return `false`?
- in which case why is `generated_at` now an optional parameter?
- also i don't see how brackets makes any difference in a chain of `&&` statements?

Reply:
yeah, you're right. I've moved the check for a nil `generated_at` into the `expired?` method - so this works properly now.